internal-messaging/message/cc/DataMessage.cc

```diff
@@ -3,96 +3,45 @@
 #include "DataMessage.h"
 #include <stdexcept>
 
-bool number_value(const char* string) {
-    bool decimal_occurred = false;
-    const int string_len = strlen(string);
-    for(int i = 0; i < string_len; ++i) {
-        if(string[i] == '.' && !decimal_occurred){ //first decimal occurrance
-            decimal_occurred = true;
-        } else if(decimal_occurred && string[i] == '.') { //any other decimal occurance
-            return false;
-        } else if(!isdigit(string[i])) { //not a number
-            if(i != 0 || string[i] != '-'){ //if the char is not setting the number as a negative
-                return false;
-            }
-        }
-    }  
-    return true; //string is a number
-}
-
 DataMessage::DataMessage()
 :Message()
 {
-    // NULL
+    flag = 0;
 }
 
 DataMessage::DataMessage(unsigned int sender, unsigned int recipient)
 :Message(sender, recipient)
 {  
-    //NULL
+    flag = 0;
 }
+
 DataMessage::DataMessage(unsigned int sender, unsigned int recipient, long time, KeyValuePairContainer contents)
 :Message(sender, recipient, time, contents)
 {
-    // NULL
+   flag = 0;
 }
 
 DataMessage::DataMessage(char* flat)
```

Comment:
Constructor is very long. Anyway to shorten this and allow for reusability in functions

Reply:
By shorten I mean break up into functions